sirepo/template/genesis.py

```diff
@@ -382,12 +414,22 @@ def _get_frame_counts(run_dir):
     return res
 
 
-def _is_text_file(path):
-    try:
-        pkio.read_text(path)
-        return True
-    except UnicodeDecodeError:
-        return False
+def _parse_maginfile(filepath):
+    if not pkio.is_pure_text(filepath):
+        raise AssertionError(f"{filepath.basename} for maginfile should be text file")
+    p = []
+    u = 1
+    with pkio.open_text(filepath) as f:
+        for line in f:
+            row = line.split()
+            if row:
+                if row[0] == _MAGIN_PLOT_FIELD:
+                    p.append(row[1])
+                if row[0] == "?" and "UNITLENGTH" in row[1]:
+                    u = row[2]
+    if p:
```

Comment:
This fails if there is a space after UNITLENGTH, for example, if the data file has this header:
```
? UNITLENGTH = 0.032
```
vs 
```
? UNITLENGTH= 0.032
```

Reply:
Oh, yeah. Thanks for catching this